### ISSUE TITLE
Regression in MinGit 2.34: Need to specify how to reconcile divergent branches

### ISSUE THREAD

**Issue by luhansen-ms** [git-for-windows/git#3739]
- [x] I was not able to find an [open](https://github.com/git-for-windows/git/issues?q=is%3Aopen) or [closed](https://github.com/git-for-windows/git/issues?q=is%3Aclosed) issue matching what I'm seeing

### Setup

 - Which version of Git for Windows are you using? Is it 32-bit or 64-bit?

```
$ git --version --build-options

** git version 2.34.1.windows.1 (64 bit) ** (note: this only impacts MinGit)
```

 - Which version of Windows are you running? Vista, 7, 8, 10? Is it 32-bit or 64-bit?

```
$ cmd.exe /c ver

** 10.0.22000.556 **
```

 - What options did you set as part of the installation? Or did you choose the
   defaults?

```
# One of the following:
> type "C:\Program Files\Git\etc\install-options.txt"
> type "C:\Program Files (x86)\Git\etc\install-options.txt"
> type "%USERPROFILE%\AppData\Local\Programs\Git\etc\install-options.txt"
$ cat /etc/install-options.txt

** n/a: mingit **
```

 - Any other interesting things about your environment that might be related
   to the issue you're seeing?

** Neither system-level Git nor system-level gitconfigs are installed. MinGit is impacted.**

### Details

 - Which terminal/shell are you running Git from? e.g Bash/CMD/PowerShell/other

** cmd/ps **

 - What commands did you run to trigger this issue? If you can provide a
   [Minimal, Complete, and Verifiable example](http://stackoverflow.com/help/mcve)
   this will help us understand the issue.

```
** git pull ** (with default gitconfig shipped with MinGit)
```
 - What did you expect to occur after running these commands?

```
Auto-merging Sandbox/Program.cs
CONFLICT (content): Merge conflict in Sandbox/Program.cs
Automatic merge failed; fix conflicts and then commit the result.
```

 - What actually happened instead?

```
hint: You have divergent branches and need to specify how to reconcile them.
hint: You can do so by running one of the following commands sometime before
hint: your next pull:
hint:
hint:   git config pull.rebase false  # merge (the default strategy)
hint:   git config pull.rebase true   # rebase
hint:   git config pull.ff only       # fast-forward only
hint:
hint: You can replace "git config" with "git config --global" to set a default
hint: preference for all repositories. You can also pass --rebase, --no-rebase,
hint: or --ff-only on the command line to override the configured default per
hint: invocation.
fatal: Need to specify how to reconcile divergent branches.
```

 - If the problem was occurring with a specific repository, can you provide the
   URL to that repository to help us with testing?

** occurs in any repo where remote branch diverges **

Further Description:

The Git for Windows 2.35 Installer Provides the following option when installing:
![image](https://user-images.githubusercontent.com/54725543/158266077-808541cd-3e72-44db-aa95-459cc5dd0ee5.png)

This seems to set the pull.rebase setting to the default value to false/merge.

In MinGit, the bundled .gitconfig does not have this default value set in any release.

In MinGit 2.33, the default value (merge) is inferred without it being set in a config file. 

In MinGit 2.34/2.35, the default value is unset, and therefore the "Need to specify how to reconcile divergent branches." message is thrown.

### Proposed Fix:
In MinGit, the default value should continue to be inferred when there is no gitconfig file specifying a default pull behavior.

or

MinGit should ship with a default `pull.rebase` value specified in `etc\gitconfig`

**Comment by dscho** [comment]
> In MinGit 2.33, the default value (merge) is inferred without it being set in a config file.

As far as I can tell, this was considered a bug and was fixed in 031e2f7ae195069d00d21cde906fce5b0318dbdd, which was already in v2.33.1.

> ### Proposed Fix:
> 
> In MinGit, the default value should continue to be inferred when there is no gitconfig file specifying a default pull behavior.
> 
> or
> 
> MinGit should ship with a default `pull.rebase` value specified in `etc\gitconfig`

The very idea that this is a bug that needs to be fixed was rejected by the Git project via 031e2f7ae195069d00d21cde906fce5b0318dbdd.

It _has_ to be the decision of the user, not of the project, according to the Git project.

If MinGit is used in the presence of Git for Windows, where the user had a chance to configure this via the installer, MinGit will respect it (by virtue of including the system config of Git for Windows when it is installed to `C:\Program Files\Git`).

If MinGit is used without any installed Git for Windows, the user had no chance to configure this, and therefore will have to do that somehow, and it is the responsibility of the MinGit user to offer a UI for that.

**Comment by luhansen-ms** [comment]
Thanks for finding this @dscho! Given this context, I definitely agree with the direction the Git project is taking this, and that MinGit users will need to offer a UI to reconcile this. 

Luke
